web/ui_utils.js

```diff
@@ -144,7 +144,7 @@ function scrollIntoView(element, spot, skipOverflowHiddenElements = false) {
  */
 function watchScroll(viewAreaElement, callback) {
   let debounceScroll = function(evt) {
-    if (rAF) {
+    if (rAF && typeof window === 'undefined') {
```

Comment:
I really don't think it's acceptable to run this check *every* time that scrolling happens in the default viewer, just so that you can support a special environment (Angular) which is apparently providing incomplete/incorrect `window` polyfills.